test/jl/tests.jl

```diff
@@ -142,7 +142,7 @@ function test6()
    println("test 6: Conversion of images from Images.jl to OpenCV Mat")
    filename = joinpath(Pkg.dir("OpenCV"), "./test/images/mandrill.jpg")
    image = imread(filename)
-   dst = Mat(int(rows(image)), int(cols(image)), CV_8UC1)
+   dst = Mat(Int(round(rows(image))), Int(round(cols(image))), CV_8UC1)
    dst = convertToMat(image)
 end
 
```

Comment:
I know what is going on.  You need to provide at least one input with a T declaration from which the C++ compiler can use to infer the type.  For example: 

If you declare the functions like this:
```julia
template <typename T1>
       inline T1 ptr_val(cv::Mat &img, int row, int col)
       {
         return(static_cast<T1>(img.at<T1>(row, col)));
       } 

template <typename T2>
       inline void ptr_val3(cv::Mat &img, int row, int col, std::vector<T2> vec)
       {
           for(int i = 0; i < 3; ++i)
           {
               img.at<cv::Vec<T2, 3>>(row, col)[i] = static_cast<T2>(vec[i]);
           }
       }

// This works
julia> img = Mat(500,500, CV_8UC3)
(class cv::Mat)
julia> @cxx ptr_val(img, 10, 10, 255)
julia> @cxx ptr_val(img, 10, 10, 233.4)
julia> @cxx ptr_val(img, 10, 10, 233.4344343)

// This also works
julia> vec1 = tostdvec([1,2])
(class std::__1::vector<long, class std::__1::allocator<long> >)
julia> @cxx ptr_val3(img, 10, 10, vec1)
julia> vec1 = tostdvec([1,2])
```

Without these changes, I get as expected errors like this:
```julia
@cxx ptr_val(img, 10, 10)
:1:1: error: no matching function for call to 'ptr_val'
^
__cxxjl_87.cpp:3:11: note: candidate template ignored: couldn't infer template argument 'T1'
inline T1 ptr_val(cv::Mat &img, int row, int col)
          ^
__cxxjl_87.cpp:9:13: note: candidate function template not viable: requires 4 arguments, but 3 were provided
inline void ptr_val(cv::Mat &img, int row, int col, T2 val)
```
However, this is surely not the goal - just an illustration of type inference.
The goal here is that the compiler infers the T type based on the Mat type, in other words, we need to use  `cv::Mat<_Tp>` as illustrated in the OpenCV [docs](http://docs.opencv.org/3.1.0/df/dfc/classcv_1_1Mat__.html).